analyzers/src/SonarAnalyzer.Common/SymbolicExecution/SymbolicConstraint.cs

```diff
@@ -18,10 +18,16 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution
 {
     public abstract class SymbolicConstraint
     {
+        protected SymbolicConstraint(ConstraintKind kind) =>
+            Kind = kind;
+
+        public ConstraintKind Kind { get; }
         public abstract SymbolicConstraint Opposite { get; }
         protected abstract string Name { get; }
```

Comment:
> Now all contraints have a Kind, has Name became obsolete?

https://github.com/SonarSource/sonar-dotnet/pull/6148#issuecomment-1304465241

Yes. `Name` should be removed.